File: csharp/src/Blockchain.Abstractions/Activities/IBlockchainActivities.cs
using Temporalio.Activities;
using Train.Solver.Blockchain.Abstractions.Models;

namespace Train.Solver.Blockchain.Abstractions.Activities;

public interface IBlockchainActivities
{
    [Activity]
    Task<BalanceResponse> GetBalanceAsync(BalanceRequest request);

    [Activity]
    Task<BlockNumberResponse> GetLastConfirmedBlockNumberAsync(BaseRequest request);

    [Activity]
    Task<bool> ValidateAddLockSignatureAsync(AddLockSignatureRequest request);

    [Activity]
    Task<HTLCBlockEventResponse> GetEventsAsync(EventRequest request);
}

File: js/src/Blockchain/Blockchain.Abstraction/Interfaces/IBlockchainActivities.ts
import { BalanceRequest } from "../Models/BalanceRequestModels/BalanceRequest";
import { BalanceResponse } from "../Models/BalanceRequestModels/BalanceResponse";
import { BaseRequest } from "../Models/BaseRequest";
import { BlockNumberResponse } from "../Models/BlockNumberResponse";
import { HTLCBlockEventResponse } from "../Models/EventModels/HTLCBlockEventResposne";
import { EventRequest } from "../Models/EventRequest";
import { AddLockSignatureRequest } from "../Models/TransactionBuilderModels/AddLockSignatureRequest";
import { TransactionBuilderRequest } from "../../Blockchain.Abstraction/Models/TransactionBuilderModels/TransactionBuilderRequest";
import { PrepareTransactionResponse } from "../../Blockchain.Abstraction/Models/TransactionBuilderModels/TransferBuilderResponse";

export interface IBlockchainActivities {
    BuildTransaction(request: TransactionBuilderRequest): Promise<PrepareTransactionResponse>;

    GetBalance(request: BalanceRequest): Promise<BalanceResponse>;  
  
    GetLastConfirmedBlockNumber(request: BaseRequest): Promise<BlockNumberResponse>;  
    
    ValidateAddLockSignature(request: AddLockSignatureRequest): Promise<boolean>;
    
    GetEvents(request: EventRequest): Promise<HTLCBlockEventResponse>;
  }